addons/arsenal/functions/fnc_onSelChangedRight.sqf

```diff
@@ -69,7 +69,14 @@ switch (_currentItemsIndex) do {
     // Secondary weapon
     case IDX_CURR_SECONDARY_WEAPON_ITEMS: {
         private _currentItemInSlot = (GVAR(currentItems) select IDX_CURR_SECONDARY_WEAPON_ITEMS) select _itemIndex;
-        private _isDisposable = CBA_disposable_replaceDisposableLauncher && {!isNil {CBA_disposable_loadedLaunchers getVariable (secondaryWeapon GVAR(center))}};
+        private _isDisposable = CBA_disposable_replaceDisposableLauncher && {!isNil "CBA_disposable_loadedLaunchers"} &&
+            {
+                if (typeName CBA_disposable_loadedLaunchers == "HASHMAP") then { // after cba 3.18
+                    (secondaryWeapon GVAR(center)) in CBA_disposable_loadedLaunchers
+                } else {
+                    (secondaryWeapon GVAR(center)) in allVariables CBA_disposable_loadedLaunchers
```

Comment:
Your solution is slightly faster with just CBA + ACE, but when you start loading mods, it gradually becomes slower.
```suggestion
                    !isNil {CBA_disposable_loadedLaunchers getVariable (secondaryWeapon player)}
```

In 2.18 we can change it to be presumably even faster with the new `isNil` syntax.

Reply:
Right now it's just `Namespace, Object, Group or HashMap` (not location)
shame because it would allow us to use same code for both hash types
but this only runs in arsenal so it shouldn't really matter

Reply:
I asked in the Arma discord if they are willing to add support for `Location`, but I won't hold my breath.